src/diffusers/models/transformers/transformer_mochi.py

```diff
@@ -142,22 +415,25 @@ def forward(
             hidden_states=norm_hidden_states,
             encoder_hidden_states=norm_encoder_hidden_states,
             image_rotary_emb=image_rotary_emb,
+            attention_mask=encoder_attention_mask,
         )
 
-        hidden_states = hidden_states + self.norm2(attn_hidden_states) * torch.tanh(gate_msa).unsqueeze(1)
-        norm_hidden_states = self.norm3(hidden_states) * (1 + scale_mlp.unsqueeze(1))
+        hidden_states = hidden_states + self.norm2(attn_hidden_states, torch.tanh(gate_msa).unsqueeze(1))
+        norm_hidden_states = self.norm3(hidden_states, (1 + scale_mlp.unsqueeze(1).to(torch.float32)))
```

Comment:
I think I'm probably missing something here, but can you point me to where do we handle the downcast for this upcast? Because otherwise all successive computations would be in float32, no?

Reply:
Downcast here
https://github.com/huggingface/diffusers/blob/b9043256273abcbdf098791a5810639c5622b26b/src/diffusers/models/transformers/transformer_mochi.py#L53

Reply:
I think it will automtically upcast, no? here in `hidden_states = hidden_states * scale` if the hidden_states is in float32 an scale is in bfloat16, the operation will be in float32 and we do not need to explicitly upcast scale, no?

this is non-merge blocking! just curious if I misunderstood something 

```python
    def forward(self, hidden_states, scale=None):
        ....
        if scale is not None:
            hidden_states = hidden_states * scale

        hidden_states = hidden_states.to(hidden_states_dtype)

        return hidden_states
```

Reply:
It's the 1 + scale. The addition has to happen FP32 otherwise we get numerical differences.